providers/provider_default_test.go

```diff
@@ -75,32 +75,86 @@ func TestProviderDataAuthorize(t *testing.T) {
 		name          string
 		allowedGroups []string
 		groups        []string
+		url           *url.URL
 		expectedAuthZ bool
 	}{
 		{
 			name:          "NoAllowedGroups",
 			allowedGroups: []string{},
 			groups:        []string{},
+			url: &url.URL{
+				Scheme: "http",
+				Host:   "my.test.idp",
+				Path:   "/",
+			},
 			expectedAuthZ: true,
 		},
 		{
 			name:          "NoAllowedGroupsUserHasGroups",
 			allowedGroups: []string{},
 			groups:        []string{"foo", "bar"},
+			url: &url.URL{
+				Scheme: "http",
+				Host:   "my.test.idp",
+				Path:   "/",
+			},
 			expectedAuthZ: true,
 		},
 		{
 			name:          "UserInAllowedGroup",
 			allowedGroups: []string{"foo"},
 			groups:        []string{"foo", "bar"},
+			url: &url.URL{
+				Scheme: "http",
+				Host:   "my.test.idp",
+				Path:   "/",
+			},
 			expectedAuthZ: true,
 		},
 		{
 			name:          "UserNotInAllowedGroup",
 			allowedGroups: []string{"bar"},
 			groups:        []string{"baz", "foo"},
+			url: &url.URL{
+				Scheme: "http",
+				Host:   "my.test.idp",
+				Path:   "/",
+			},
 			expectedAuthZ: false,
 		},
+		{
+			name:          "NoAllowedGroupsMatchingPath",
+			allowedGroups: []string{"bar", "/path|doo"},
+			groups:        []string{"foo"},
+			url: &url.URL{
+				Scheme: "http",
+				Host:   "my.test.idp",
+				Path:   "/path",
+			},
+			expectedAuthZ: false,
+		},
+		{
+			name:          "AllowedGroupsMatchingPath",
+			allowedGroups: []string{"bar", "/path|foo"},
+			groups:        []string{"foo"},
+			url: &url.URL{
+				Scheme: "http",
+				Host:   "my.test.idp",
+				Path:   "/path",
+			},
+			expectedAuthZ: true,
+		},
+		{
+			name:          "MoreSpecificPathMatches",
```

Comment:
Equating "more specific" to "longer prefix match" is ambiguous and potentially misleading?

Which is "more specific" - `/administration` or `/admin/a/b`? I think most people would say the latter, because it is 'deeper'. But given a rule for `/administration` and a rule for `/admin/a/b`, your code will pick the former as being "more specific" and hence the rule that is used.

An alternative to consider would be to use the first match in the list?

(I'm not associated with the project, this is a drive-by comment!)

Reply:
This makes sense indeed. Maybe we should allow globing or regex. It’s more cognitive load on the user but at the same time the contract is clear.

Reply:
I can imagine some use cases might benefit from regex/glob matching. But I think it's an orthogonal question - you could still have multiple matches and need a way to pick one, i.e. first wins.

Reply:
I see what you mean. Probably first wins is more intuitive.